ide/db/src/org/netbeans/modules/db/explorer/DbSchemasQuery.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.db.explorer;
+
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.netbeans.api.db.explorer.DatabaseException;
+import org.openide.util.Exceptions;
+
+/**
+ * 
+ * @author Jan Horvath
+ */
+public class DbSchemasQuery {
+
+    /**
+     * Retrieves a list of database schemas available for given connection.
+     * 
+     * @param dbconn
+     * @return List of schemas or null if connection failed
+     */
+    public static List<String> getSchemasForConnection(org.netbeans.api.db.explorer.DatabaseConnection dbconn) {
```

Comment:
Please use the metadata API. See `org.netbeans.modules.db.explorer.node.ConnectionNode.updateModel()` to see how the metamodel can be accessed and `org.netbeans.modules.db.explorer.node.SchemaNodeProvider.initialize()` for a usage sample.